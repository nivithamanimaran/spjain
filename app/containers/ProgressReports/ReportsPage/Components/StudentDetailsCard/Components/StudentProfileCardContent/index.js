import React from 'react';
import { Grid, makeStyles, Box, Typography, Avatar } from '@material-ui/core';
import { useTheme } from '@material-ui/core';
import KenAvatar from '../../../../../../../components/KenAvatar';

const useStyles = makeStyles(theme => ({
  dobContainer: {
    textAlign: 'center',
  },
  text: {
    color: theme.palette.KenColors.neutral103,
    fontWeight: 'bold',
  },
  headings: {
    color: theme.palette.KenColors.neutral61,
  },
  box: {
    flexGrow: 1,
  },
  innerBox: {
    textAlign: 'left',
  },
  name: {
    fontSize: 14,
    color: theme.palette.KenColors.neutral900,
    fontWeight: '600',
  },
  avatar: {
    width: 32,
    height: 32,
  },
}));

const StudentProfileCardContent = props => {
  const { data = [] } = props;
  const classes = useStyles();
  //   const { t } = useTranslation();
  const theme = useTheme();

  return (
    <Box className={classes.box} data-testid="student-profile-card-content">
      <Grid container spacing={2}>
        <Box data-testid="student-profile-avatar" pb={2}>
          <Grid container spacing={1} direction="row" alignItems="center">
            <Grid item>
              <KenAvatar
                value={data ? data.name && data.name[0] : ''}
                className={classes.avatar}
              />
            </Grid>
            <Grid item>
              <Typography className={classes.name}>{data.name}</Typography>
            </Grid>
          </Grid>
        </Box>
        {data &&
          Array.isArray(data.details) &&
          data.details.map((item, index) => {
            return (
              <>
                {item.label && (
                  <Grid item xs={12} key={index}>
                    <Box className={classes.innerBox}>
                      <div data-testid="student-profile-card-content-label">
                        <Typography
                          variant="body2"
                          className={classes.headings}
                        >
                          {item.label}
                        </Typography>
                      </div>
                      <Box mt={1}>
                        <div data-testid="student-profile-card-content-value">
                          <Typography
                            className={classes.text}
                            variant="body2"
                            color="primary"
                          >
                            {item.value}
                          </Typography>
                        </div>
                      </Box>
                    </Box>
                  </Grid>
                )}
              </>
            );
          })}
        {/* <Grid item xs={12}>
          <Typography className={classes.headings} variant="body2">
            {t('headings:Reports_Class')}
          </Typography>
          <Box mt={1}>
            <Typography
              className={classes.text}
              variant="body2"
              color="primary"
            >
              {standard}
            </Typography>
          </Box>
        </Grid>
        <Grid item xs={6}>
          <Typography variant="body2" className={classes.headings}>
            {t('headings:Reports_Admission_Number')}
          </Typography>
          <Box mt={1}>
            <Typography
              className={classes.text}
              variant="body2"
              color="primary"
            >
              {admissionNumber}
            </Typography>
          </Box>
        </Grid>
        <Grid item xs={6} className={classes.dobContainer}> 
          <Typography
            className={classes.items}
            variant="body2"
            className={classes.headings}
          >
            {t('headings:Reports_DOB')}
          </Typography>
          <Box mt={1}>
            <Typography
              className={classes.text}
              variant="body2"
              color="primary"
            >
              {dob}
            </Typography>
          </Box>
        </Grid>*/}
      </Grid>
    </Box>
  );
};
export default StudentProfileCardContent;
