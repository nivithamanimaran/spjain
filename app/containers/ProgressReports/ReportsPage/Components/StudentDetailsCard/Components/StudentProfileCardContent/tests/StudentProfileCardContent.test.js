import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import StudentProfileCardContent from '../index';
import ThemeProvider from '../../../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Student-Details-Card-Content-Page', () => {
  it('Student Details Card Content renders correctly without props', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardContent />
        </Router>
      </ThemeProvider>
    );
    const content = getByTestId('student-profile-card-content');
    expect(content).toBeTruthy();
    expect(content.textContent).toBe('');

    expect(getByTestId('student-profile-card-content')).toMatchSnapshot();
  });
  it('Student Details Card Content renders correctly with props', () => {
    const div = document.createElement('div');
    const data = [{ label: 'Class', value: 'Montessori' }];

    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardContent data={data} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentProfileCardContent data={data} />
        </Router>
      </ThemeProvider>
    );

    const cardContent = queryByTestId('student-profile-card-content');
    expect(cardContent).toBeTruthy();

    const label = queryByTestId('student-profile-card-content-label');
    expect(label).toBeTruthy();
    expect(label.textContent).toBe('Class');

    const value = queryByTestId('student-profile-card-content-value');
    expect(value).toBeTruthy();
    expect(value.textContent).toBe('Montessori');
  });
});
