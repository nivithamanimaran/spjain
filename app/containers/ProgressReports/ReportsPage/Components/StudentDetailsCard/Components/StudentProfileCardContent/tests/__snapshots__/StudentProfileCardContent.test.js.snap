// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student-Details-Card-Content-Page Student Details Card Content renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-7 makeStyles-box-4"
  data-testid="student-profile-card-content"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  />
</div>
`;
